src/watertap_contrib/reflo/unit_models/tests/test_multi_effect_crystallizer.py

```diff
@@ -0,0 +1,1127 @@
+    first_effect.heating_steam[0].dh_vap_mass
+    first_effect.heating_steam.calculate_state(
+        var_args={
+            ("flow_mass_phase_comp", ("Liq", "H2O")): 0,
+            ("pressure", None): saturated_steam_pressure,
+            ("pressure_sat", None): saturated_steam_pressure,
+        },
+        hold_state=True,
+    )
+    first_effect.heating_steam[0].flow_mass_phase_comp["Vap", "H2O"].unfix()
+
+    return m
+
+
+def build_mec3():
+
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+
+    m.fs.properties = NaClParameterBlock()
+    m.fs.vapor_properties = WaterParameterBlock()
+
+    m.fs.unit = mec = MultiEffectCrystallizer(
+        property_package=m.fs.properties,
+        property_package_vapor=m.fs.vapor_properties,
+        number_effects=3,
+    )
+
+    feed_flow_mass = 10
+    feed_mass_frac_NaCl = 0.25
+    crystallizer_yield = 0.55
+    operating_pressures = [0.45, 0.25, 0.208]
+    feed_mass_frac_H2O = 1 - feed_mass_frac_NaCl
+
+    atm_pressure = 101325 * pyunits.Pa
+    saturated_steam_pressure_gage = 3.5 * pyunits.bar
+    saturated_steam_pressure = atm_pressure + pyunits.convert(
+        saturated_steam_pressure_gage, to_units=pyunits.Pa
+    )
+
+    for (_, eff), op_pressure in zip(mec.effects.items(), operating_pressures):
+        eff.effect.properties_in[0].flow_mass_phase_comp["Liq", "H2O"].fix(
+            feed_flow_mass * feed_mass_frac_H2O
+        )
+        eff.effect.properties_in[0].flow_mass_phase_comp["Liq", "NaCl"].fix(
+            feed_flow_mass * feed_mass_frac_NaCl
+        )
+        eff.effect.properties_in[0].pressure.fix(feed_pressure)
+        eff.effect.properties_in[0].temperature.fix(feed_temperature)
+
+        eff.effect.properties_in[0].flow_mass_phase_comp["Sol", "NaCl"].fix(eps)
+        eff.effect.properties_in[0].flow_mass_phase_comp["Vap", "H2O"].fix(eps)
+        eff.effect.properties_in[0].conc_mass_phase_comp[...]
+        eff.effect.crystallization_yield["NaCl"].fix(crystallizer_yield)
+        eff.effect.crystal_growth_rate.fix()
+        eff.effect.souders_brown_constant.fix()
+        eff.effect.crystal_median_length.fix()
+        eff.effect.pressure_operating.fix(
+            pyunits.convert(op_pressure * pyunits.bar, to_units=pyunits.Pa)
+        )
+        eff.effect.overall_heat_transfer_coefficient.set_value(0.1)
+
+    first_effect = m.fs.unit.effects[1].effect
+
+    first_effect.overall_heat_transfer_coefficient.fix(0.1)
+    first_effect.heating_steam[0].dh_vap_mass
+    first_effect.heating_steam.calculate_state(
+        var_args={
+            ("flow_mass_phase_comp", ("Liq", "H2O")): 0,
+            ("pressure", None): saturated_steam_pressure,
+            ("pressure_sat", None): saturated_steam_pressure,
+        },
+        hold_state=True,
+    )
+    first_effect.heating_steam[0].flow_mass_phase_comp["Vap", "H2O"].unfix()
+
+    return m
+
+
+class TestMultiEffectCrystallizer_4Effects:
```

Comment:
So 4 effects is the default, and you test for 4 effects and 3 effects. What happens if the user tries entering 1 effect? Similarly, what about 2 effects? I went scanned through the core code quickly, so I know I missed this. I just wonder if we should just add two quick, shortened versions of these tests just to ensure that we get what we'd expect from entering 1 effect or 2 effects. For example, in 2 effects, I might expect some special handling of the first and last effect, hence interest in this case. For 1 effect, I would expect the code to handle this case or throw an exception if the multieffect model is not setup to handle specification of 1 effect.

Reply:
addressed in e0b5b44.

Added test for 2 effects.
If the user tries to use ```MultiEffectCrystallizer``` with a single effect, it will raise ```ConfigurationError``` and they will be directed to use the ```CrystallizerEffect``` model.

Reply:
Nice. One thought I have--say you wanted to run optimization cases across a range of conditions and a range of number of effects, including a single effect. You'd need to add some logic in to such a script to toggle between the CrystallizerEffect model for the case where a single effect is optimized, and then toggle to the MultiEffectCrystallizer model for all other cases. 

All that just to say, it might be useful to actually allow for a single effect via MultiEffectCrystallizer too. Definitely not something to consider for this PR--just food for thought.

Reply:
Thinking about this now, I don't think it would be very hard to implement, maybe a little clunky, but I do like the idea of having all the functionality for a unit model within a single model rather than directing the user to use a different unit model. A consideration for the future.